### ISSUE TITLE
BUG with potions and buff skills

### ISSUE THREAD

**Issue by DimasjkTV** [suriyun-production/mmorpg-kit-docs#171]
Single player game
1. We use buff for 30 seconds.
2. We use potions that increase stats by 10 minutes.
3. As a result, our first buff with 30 seconds turns into 10 minutes
[Potions BUG 1.51.zip](https://github.com/suriyun-production/mmorpg-kit-docs/files/4022492/Potions.BUG.1.51.zip)

**Comment by insthync** [comment]
Try fix it by add codes in UICharacterBuff.cs

Line 117 (In function: UpdateData)
```
            buffRemainsDuration = CharacterBuff.buffRemainsDuration;
```

**Comment by DimasjkTV** [comment]
> Try fix it by add codes in UICharacterBuff.cs
> 
> Line 117 (In function: UpdateData)
> 
> ```
>             buffRemainsDuration = CharacterBuff.buffRemainsDuration;
> ```

Works

**Comment by DimasjkTV** [comment]
but there is a little problem. Time is up, but the icon continues to display
![image](https://user-images.githubusercontent.com/47291794/71811993-1d638100-3087-11ea-91d9-e4ad5c8caee1.png)

**Comment by DimasjkTV** [comment]
This buff gives damage to the character. Damage is not updated. But in fact, the character does not have a buff effect. Bug in updating information.
![image](https://user-images.githubusercontent.com/47291794/71812061-4dab1f80-3087-11ea-9db9-aa64363c0e13.png)

**Comment by DimasjkTV** [comment]
After logout we have normal dmg and buffs
![image](https://user-images.githubusercontent.com/47291794/71812101-6582a380-3087-11ea-84c7-285d8efb0685.png)
![image](https://user-images.githubusercontent.com/47291794/71812107-6adfee00-3087-11ea-89c3-6ed20a4dd34f.png)

**Comment by insthync** [comment]
Try changes codes in BasePlayerCharacterController.cs
From Line 286 (function: OnBuffsOperation) to
```
        protected void OnBuffsOperation(LiteNetLibSyncList.Operation operation, int index)
        {
            if (PlayerCharacterEntity.IsOwnerClient && CacheUISceneGameplay != null)
                CacheUISceneGameplay.UpdateCharacter();
        }
```

**Comment by DimasjkTV** [comment]
> Try changes codes in BasePlayerCharacterController.cs
> From Line 286 (function: OnBuffsOperation) to
> 
> ```
>         protected void OnBuffsOperation(LiteNetLibSyncList.Operation operation, int index)
>         {
>             if (PlayerCharacterEntity.IsOwnerClient && CacheUISceneGameplay != null)
>                 CacheUISceneGameplay.UpdateCharacter();
>         }
> ```

Works great!

**Comment by insthync** [comment]
This fix included in 1.52
